cmd/kubelet/app/server.go

```diff
@@ -253,7 +253,11 @@ func (s *KubeletServer) Run(_ []string) error {
 		glog.Warning(err)
 	}
 
-	client, err := s.createAPIServerClient()
+	var apiclient *client.Client
+	clientConfig, err := s.CreateAPIServerClientConfig()
+	if err != nil {
```

Comment:
I'm confused about the change you made starting here. To avoid changing the existing logic, shouldn't it be

```
if err != nil {
  ...log something...
}
apiclient, err = client.New(clientconfig)
if (err != nil) {
  apiclient = nil
}
```

?